### PR TITLE
feat: replace hand-crafted protobuf OTLP exporter with opentelemetry-otlp SDK

### LINKED ISSUE #97: OTLP exporter: use standard opentelemetry-otlp SDK instead of hand-crafted protobuf
## Summary
The current OTLP exporter manually crafts protobuf messages. Replace with the standard `opentelemetry-otlp` SDK for correctness, maintainability, and spec compliance.

## Scope
- Replace hand-crafted protobuf encoding in `src/exporter/otlp.rs` with `opentelemetry-otlp` crate
- Use `opentelemetry-sdk` for metric data model
- Keep the same export behavior (HTTP transport, protobuf encoding)
- Update `Cargo.toml` dependencies
- Update tests
- Update `spec/export-otlp.md` to note SDK usage

## Why
- Standard SDK handles proto evolution, serialization edge cases, and spec compliance
- Reduces custom code to maintain
- Better alignment with OpenTelemetry ecosystem

### PR DESCRIPTION
**Pull request by r12f**
## Summary

Replaces the manual prost-based protobuf encoding in the OTLP exporter with the standard `opentelemetry-otlp` + `opentelemetry_sdk` pipeline.

## Changes

- **Cargo.toml**: Updated `opentelemetry` crates from 0.27 → 0.29 with proper feature flags (`metrics`, `http-proto`)
- **src/exporter/otlp/mod.rs**: Complete rewrite (~560 lines of hand-crafted protobuf → ~200 lines using SDK)
  - `SdkMeterProvider` with `PeriodicReader` + OTLP HTTP `MetricExporter`
  - Gauge metrics → OTel `f64_gauge` instruments
  - Counter metrics → OTel `f64_counter` instruments
  - Global labels → `Resource` attributes
  - Config `endpoint`/`headers`/`timeout`/`interval` → SDK equivalents
  - Scope name: `bus-exporter`
- **src/exporter/otlp/mod_tests.rs**: Updated tests for SDK-based implementation
- **MetricExporter trait interface**: Unchanged — no impact on rest of codebase

## Testing

- `cargo fmt` ✅
- `cargo clippy -- -D warnings` ✅  
- `cargo test` — all 228 tests pass ✅

Closes #97

### CODE REVIEW

File: src/exporter/otlp/mod.rs [hunk 1]
```diff
@@ -1,402 +1,95 @@
...
+) -> Result<SdkMeterProvider> {
+    let exporter = opentelemetry_otlp::MetricExporter::builder()
+        .with_http()
+        .with_endpoint(endpoint)
+        .with_headers(headers.clone())
+        .with_timeout(timeout)
+        .build()
+        .map_err(|e| anyhow::anyhow!("Failed to build OTLP metric exporter: {e}"))?;
+
+    let reader = opentelemetry_sdk::metrics::PeriodicReader::builder(exporter)
```
Comment (r12f) [inline src/exporter/otlp/mod.rs]: **Spec compliance concern: Retry/backoff behavior now opaque.**

The spec (`spec/export-otlp.md`) requires specific retry behavior: 1s→2s→4s→max 30s backoff, no retry on 4xx (except 429), and `Retry-After` header respect on 429. The SDK's built-in retry in the OTLP exporter may not match these requirements exactly — its backoff parameters and Retry-After handling are not configurable.

This is acceptable if you're intentionally relaxing the spec to "use SDK defaults," but if strict compliance is needed, you may want to document the deviation or configure the exporter's retry policy (if the 0.29 SDK exposes it).

File: src/exporter/otlp/mod.rs [hunk 2]
```diff
@@ -1,402 +1,95 @@
...
+        let unit = m.unit.clone();
 
         match m.metric_type {
             MetricType::Gauge => {
-                // Gauge message (field 5 of Metric)
-                let mut gauge = Vec::new();
-                encode_ld(1, &dp, &mut gauge); // data_points
-                encode_ld(5, &gauge, &mut metric);
+                let gauge = meter
+                    .f64_gauge(name)
```
Comment (r12f) [inline src/exporter/otlp/mod.rs]: **Nit: Instruments re-created every export cycle.**

Each call to `record_metrics` builds new `f64_gauge`/`f64_counter` instruments via `meter.f64_gauge(name).build()`. The SDK deduplicates by name internally so this is *correct*, but it's unnecessary allocation churn every cycle. Consider caching instruments in a `HashMap<String, _>` or building them once at startup if the metric set is known ahead of time.

File: src/exporter/otlp/mod.rs [hunk 3]
```diff
@@ -1,402 +1,95 @@
...
-                encode_ld(1, &dp, &mut sum); // data_points
-                encode_varint_field(2, 2, &mut sum); // AGGREGATION_TEMPORALITY_CUMULATIVE = 2
-                encode_varint_field(3, 1, &mut sum); // is_monotonic = true
-                encode_ld(7, &sum, &mut metric);
+                let counter = meter
+                    .f64_counter(name)
+                    .with_description(desc)
+                    .with_unit(unit)
+                    .build();
+                counter.add(m.value, &attrs);
```
[the remark this answers is not in the document]
Comment (r12f) [reply src/exporter/otlp/mod.rs]: Fixed in 64e2013 — switched to `f64_observable_counter` / `f64_observable_gauge` with callbacks that report absolute values. The SDK now handles delta computation internally, so cumulative counters are no longer misinterpreted as deltas.

File: src/exporter/otlp/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (r12f) [reply src/exporter/otlp/mod.rs]: Fixed in 64e2013 — removed the manual `force_flush()` from the `run()` loop. The `PeriodicReader` now solely handles export scheduling. For the `MetricExporter` trait impl (which needs on-demand flush), the PeriodicReader interval is set to 86400s to avoid double exports.
